react/ProductGiftsContext.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { createContext, useContext, FC } from 'react'
+import useProduct from 'vtex.product-context/useProduct'
+
+interface Props {
+  maxVisibleItems: number | 'showAll'
+}
+
+interface State extends Props {
+  gifts: Gift[]
+}
+
+const GiftsStateContext = createContext<State | undefined>(undefined)
+
+const ProductGiftsContextProvider: FC<Props> = ({
+  children,
+  maxVisibleItems,
+}) => {
+  const productContext: Maybe<ProductContextState> = useProduct()
+
+  if (!productContext) {
+    return null
+  }
+
+  const sellers = productContext?.selectedItem?.sellers
+  const gifts =
+    sellers?.reduce((acc: Gift[], curr) => {
+      return [...acc, ...(curr.commertialOffer.gifts || [])]
```

Comment:
dont do this, you are doing O(nˆ2) work. Use acc.push(...(curr.commertialOffer.gifts || []))

Reply:
Awesome!